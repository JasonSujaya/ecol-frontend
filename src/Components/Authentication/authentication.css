.formContainer {
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
}

.authenticationContainer{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 20px;
}