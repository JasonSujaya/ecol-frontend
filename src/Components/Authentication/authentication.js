import axios from "axios";

export const storeToken = (token) => {
  localStorage.token = token;
};

export const checkForToken = () => {
  let tokenValue = localStorage.token;
  if (tokenValue != null) {
    if (tokenValue.length != 0) {
      return true;
    } else {
      return false;
    }
  } else {
    return false;
  }
};

export const login = async (_email, _password) => {
  const user = {
    username: _email,
    password: _password,
  };

  const response = await axios.post(
    `http://127.0.0.1:8000/api/user/login/`,
    user,
    {
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  return response;
};

export const formsNotEmpty = (_email, _password) => {
  if (_email.length != 0 && _password.length != 0) {
    return true;
  } else {
    return false;
  }
};

export const signUp = async (_email, _first_name, _last_name, _password) => {
  const user = {
    email: _email,
    first_name: _first_name,
    last_name: _last_name,
    password: _password,
  };

  const response = await axios.post(
    `http://127.0.0.1:8000/api/user/profiles/`,
    user,
    {
      headers: {
        "Content-Type": "application/json",
      },
    }
  );

  return response;
};

export const inputNotEmpty = (_input) => {
  if (_input.length != 0) {
    return true;
  } else {
    return false;
  }
};

export const checkPassword = (password) => {
  let re = "";
  if (password != "") {
    if (password.length < 6) {
      return "Error: Password must contain at least six characters!";
    }
    re = /[0-9]/;
    if (!re.test(password)) {
      return "Error: password must contain at least one number (0-9)!";
    }
    re = /[a-z]/;
    if (!re.test(password)) {
      return "Error: password must contain at least one lowercase letter (a-z)!";
    }
    re = /[A-Z]/;
    if (!re.test(password)) {
      return "Error: password must contain at least one uppercase letter (A-Z)!";
    }
  } else {
    return "Error: Please check that you've entered and confirmed your password!";
  }

  return true;
};
