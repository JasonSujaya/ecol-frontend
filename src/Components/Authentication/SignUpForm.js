import React from "react";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import "./authentication.css";

import {
  signUp,
  login,
  inputNotEmpty,
  checkPassword,
} from "./authentication.js";

class SignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      first_name: "",
      last_name: "",
      password: "",
      validForm: false,
      invalidPassword: false,
      invalidPasswordMessage: "",
      error: false,
      signUpSuccessfull: false,
    };
  }

  onSignUp = async (e) => {
    e.preventDefault();
    this.setState({
      invalidPassword: true,
      invalidPasswordMessage: checkPassword(this.state.password),
    });

    if (checkPassword(this.state.password) == true) {
      let result = signUp(
        this.state.email,
        this.state.first_name,
        this.state.last_name,
        this.state.password
      );
      result
        .then((response) => {
          let result = login(this.state.email, this.state.password);
          result.then((response) => {
            this.props.onSuccess(response.data);
            this.setState({
              email: "",
              first_name: "",
              last_name: "",
              password: "",
            });
          });
        })
        .catch((error) => {
          this.setState({ error: true });
        });
    } else {
      this.setState({
        invalidPassword: true,
        invalidPasswordMessage: checkPassword(this.state.password),
      });
    }
  };

  checkValidForm = (e) => {
    let validity =
      inputNotEmpty(this.state.email) &&
      inputNotEmpty(this.state.password) &&
      inputNotEmpty(this.state.first_name) &&
      inputNotEmpty(this.state.last_name);
    this.setState({ validForm: validity });
  };

  onErrorDisplayMessage = () => {
    if (this.state.error) {
      return (
        <div className="signUpError">You have an existing account already!</div>
      );
    }
  };

  invalidPasswordDisplayMessage = () => {
    if (this.state.invalidPasswordMessage) {
      return (
        <div className="invalidPasswordError">
          {this.state.invalidPasswordMessage}
        </div>
      );
    }
  };

  render() {
    return (
      <Grid container direction="column" justify="center" alignItems="center">
        <form noValidate autoComplete="off">
          <div className="formContainer">
            <TextField
              required
              className="emailInput"
              id="standard-required"
              label="email"
              onChange={(e) =>
                this.setState(
                  {
                    email: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.email}
            />
            <TextField
              id="standard-password-input"
              className="passwordInput"
              label="password"
              type="password"
              required
              onChange={(e) =>
                this.setState(
                  {
                    password: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.password}
            />
            <TextField
              required
              id="standard-required"
              className="firstNameInput"
              label="Your First Name"
              onChange={(e) =>
                this.setState(
                  {
                    first_name: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.first_name}
            />
            <TextField
              required
              id="standard-required"
              label="Your Last Name"
              className="lastNameInput"
              onChange={(e) =>
                this.setState(
                  {
                    last_name: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.last_name}
            />
            <Button
              variant="contained"
              onClick={this.onSignUp}
              disabled={!this.state.validForm}
              className="signUpButton"
            >
              Sign Up
            </Button>
          </div>
        </form>
        {this.onErrorDisplayMessage()}
        {this.invalidPasswordDisplayMessage()}
      </Grid>
    );
  }
}

export default SignUpForm;
