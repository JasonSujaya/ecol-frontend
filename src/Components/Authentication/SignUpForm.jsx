import React from "react";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import { signUp, inputNotEmpty, checkPassword } from "./authentication.js";

const styles = (theme) => ({});

class SignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      first_name: "",
      last_name: "",
      password: "",
      validForm: false,
    };
  }

  onSignUp = (e) => {
    e.preventDefault();
    if (checkPassword(e)) {
      signUp(
        this.state.email,
        this.state.first_name,
        this.state.last_name,
        this.state.password
      );
    } else {
      console.log(checkPassword(e));
    }
    this.setState({ email: "", first_name: "", last_name: "", password: "" });
  };

  checkValidForm = (e) => {
    let validity =
      inputNotEmpty(this.state.email) &&
      inputNotEmpty(this.state.password) &&
      inputNotEmpty(this.state.first_name) &&
      inputNotEmpty(this.state.last_name);
    this.setState({ validForm: validity });
  };

  render() {
    return (
      <Grid container direction="column" justify="center" alignItems="center">
        <form noValidate autoComplete="off">
          <div>
            <TextField
              required
              className="emailInput"
              id="standard-required"
              label="email"
              onChange={(e) =>
                this.setState(
                  {
                    email: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.email}
            />
            <TextField
              id="standard-password-input"
              className="passwordInput"
              label="password"
              type="password"
              required
              onChange={(e) =>
                this.setState(
                  {
                    password: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.password}
            />
            <TextField
              required
              id="standard-required"
              className="firstNameInput"
              label="Your First Name"
              onChange={(e) =>
                this.setState(
                  {
                    first_name: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.first_name}
            />
            <TextField
              required
              id="standard-required"
              label="Your Last Name"
              className="lastNameInput"
              onChange={(e) =>
                this.setState(
                  {
                    last_name: e.target.value,
                  },
                  () => this.checkValidForm()
                )
              }
              value={this.state.last_name}
            />
            <Button
              variant="contained"
              onClick={this.onSignUp}
              disabled={!this.state.validForm}
              className="signUpButton"
            >
              Sign Up
            </Button>
          </div>
        </form>
      </Grid>
    );
  }
}

export default SignUpForm;
