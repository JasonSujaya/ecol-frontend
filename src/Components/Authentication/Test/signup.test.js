import axios from "axios";
import { shallow, mount } from "enzyme";

import SignUpForm from "../SignUpForm.jsx";
import { checkPassword, signUp } from "../authentication.js";

function resolvePromiseOnSimulate() {
  return new Promise((resolve) => {
    setTimeout(resolve, 0);
  });
}

const setUp = () => {
  axios.post.mockResolvedValue({
    data: {
      first_name: "Jason",
      last_name: "BackName",
      id: 1,
      email: "mock@gmail.com",
    },
  });
};

describe("Sign Up component", () => {
  it("Sign up renders correctly", () => {
    const wrapper = shallow(<SignUpForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it("When input is empty, button is not clickable", async () => {
    // Act
    const wrapper = shallow(<SignUpForm />);
    let buttonElement = wrapper.find(".signUpButton");

    // Assert
    expect(wrapper.state("validForm")).toBeFalsy;
  });

  it("When input is not empty, button is clickable", async () => {
    // Act
    const wrapper = shallow(<SignUpForm />);
    wrapper
      .find(".emailInput")
      .simulate("change", { target: { value: "foo" } });
    wrapper
      .find(".passwordInput")
      .simulate("change", { target: { value: "foo" } });
    wrapper
      .find(".firstNameInput")
      .simulate("change", { target: { value: "foo" } });
    wrapper
      .find(".lastNameInput")
      .simulate("change", { target: { value: "foo" } });

    let buttonElement = wrapper.find(".signUpButton");

    // Assert
    expect(wrapper.state("validForm")).toBeFalsy;
    expect(buttonElement.prop("disabled")).toBeTruthy;
  });
});

describe("Sign Up Post Request", () => {
  it("Returns User data token when the sign in is successful ", async () => {
    // Arrange
    setUp();

    // Act
    const response = await signUp(
      "mock@email.com",
      "mockpassword",
      "firs_name",
      "last_name"
    );

    // Assert
    // expect(response.data.first_name).toEqual("Jason");
    expect(response.data.email).toEqual("mock@gmail.com");
  });
});

describe("Password Unit Test", () => {
  it("Check if password is empty ", () => {
    expect(checkPassword("")).toEqual(
      "Error: Please check that you've entered and confirmed your password!"
    );
  });
  it("Check if password is at least six character", () => {
    expect(checkPassword("Hello")).toEqual(
      "Error: Password must contain at least six characters!"
    );
    expect(checkPassword("     ")).toEqual(
      "Error: Password must contain at least six characters!"
    );
  });

  it("Check if password contains number", () => {
    expect(checkPassword("Hellooooo")).toEqual(
      "Error: password must contain at least one number (0-9)!"
    );
  });

  it("Check if password contains at least one lower chracter", () => {
    expect(checkPassword("123456")).toEqual(
      "Error: password must contain at least one lowercase letter (a-z)!"
    );

    expect(checkPassword("123456A")).toEqual(
      "Error: password must contain at least one lowercase letter (a-z)!"
    );
  });
  it("Check if password contains at least one uppercase character", () => {
    expect(checkPassword("abcdefg1")).toEqual(
      "Error: password must contain at least one uppercase letter (A-Z)!"
    );

    expect(checkPassword("123456a")).toEqual(
      "Error: password must contain at least one uppercase letter (A-Z)!"
    );
  });

  it("Checks if valid password returns OK", () => {
    expect(checkPassword("1ValidPassword")).toEqual(true);
  });
});
