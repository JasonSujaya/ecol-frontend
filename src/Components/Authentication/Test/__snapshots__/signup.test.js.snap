// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up component Sign up renders correctly 1`] = `ShallowWrapper {}`;
